src/revpi-factory-reset

```diff
@@ -54,30 +65,37 @@ if [ "$last_part_size" -lt "$last_part_max" ] ; then
 	/sbin/resize2fs "/dev/$last_part"
 fi
 
-/bin/sed --follow-symlinks -r -i -e "/^dtoverlay=revpi-($full_devicetype_list)/d" /boot/config.txt
-echo "dtoverlay=revpi-$ovl" >> /boot/config.txt
-/usr/bin/dtoverlay "revpi-$ovl"
-/bin/cp "/boot/overlays/revpi-$ovl-dt-blob.dtbo" /boot/dt-blob.bin 2>/dev/null
+
+if ! "$with_hat_eeprom"; then
+	/bin/sed --follow-symlinks -r -i -e "/^dtoverlay=revpi-($full_devicetype_list)/d" /boot/config.txt
```

Comment:
Why don't you us the same sed for both cases? And then just put it above your if?

Reply:
to avoid the use of $full_devicetype_list in with hat-eeprom case.

Reply:
But why is it important to use `$full_devicetype_list` in one case and not in the other?